### PR TITLE
Add two-factor authentication for superusers.

### PR DESCRIPTION
**Pull request by dracos**
Todo:
- [x] test of turning on/off 2FA
- [x] test superuser being denied report making by code, and go through 2FA login flow

### CODE REVIEW

File: templates/web/base/auth/2faform.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[% INCLUDE 'header.html', bodyclass = 'fullwidthpage', title = loc('Confirm account') %]
+
+  <div class="confirmation-header confirmation-header--phone">
+    [% IF incorrect_code %]
+        <h1>[% loc('Sorry, that wasn&rsquo;t the correct code') %]</h1>
+        <p>[% loc('Try again') %]:</p>
+    [% ELSE %]
+        <h1>[% loc("Nearly done! Now check your phone&hellip;") %]</h1>
+        <p>[% loc("Please generate a 2FA code and enter it below:") %]</p>
```
Comment (struan) [inline templates/web/base/auth/2faform.html]: Not sure that's the correct title to use :)

File: templates/web/base/auth/generate_token.html [hunk 2]
```diff
@@ -31,6 +52,7 @@ <h1>[% loc('Generate token') %]</h1>
 
     <p>
     <input name="generate_token" type="submit" class="btn" value="[% existing_token ? loc('Replace token') : loc('Generate token') %]">
+    <input name="toggle_2fa" type="submit" class="btn" value="[% has_2fa ? loc('Deactivate 2FA') : loc('Activate 2FA') %]">
```
Comment (struan) [inline templates/web/base/auth/generate_token.html]: Should we not only show this button to superusers?

File: templates/web/base/auth/generate_token.html [hunk 3]
```diff
@@ -12,12 +12,33 @@ <h1>[% loc('Your token has been generated') %]</h1>
         <span>[% existing_token | html %]</span>
     </p>
 
+<img src="[% qr_code %]">
```
Comment (struan) [inline templates/web/base/auth/generate_token.html]: Will this ever appear here?

File: perllib/FixMyStreet/App/Controller/Auth.pm [hunk 4]
```diff
@@ -232,6 +232,10 @@ sub process_login : Private {
     $c->detach( '/page_error_403_access_denied', [] )
         if FixMyStreet->config('SIGNUPS_DISABLED') && !$user->in_storage && !$data->{old_user_id};
 
+    # Superusers using 2FA can not log in by code
```
Comment (struan) [inline perllib/FixMyStreet/App/Controller/Auth.pm]: There seems to be some inconsistency between this and creating a new report which doesn't allow logging in by code at all for super users. I think I'd be fine with preventing it everywhere.

File: templates/web/base/auth/2faform.html (hunk 1 above)
Comment (dracos) [reply templates/web/base/auth/2faform.html]: I assumed most people will have the Authenticator app on their phone, so that is enough right for me :)

**Comment by dracos** [comment]
I've changed it so superuser-with-2FA-on can not log in by email/text, but to make up for that they *can* now log in by password/2FA whilst making a report/update, with a variant on the FB/Twitter way of remembering where they are in the flow.

**Pull request merged**